test/Datadog.Trace.Security.IntegrationTests/AspNetBase.cs

```diff
@@ -0,0 +1,200 @@
+// <copyright file="AspNetBase.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.Security.IntegrationTests
+{
+    public class AspNetBase : TestHelper
+    {
+        private readonly HttpClient httpClient;
+        private int httpPort;
+        private Process process;
+
+        public AspNetBase(string sampleName, ITestOutputHelper outputHelper, string samplesDir = null)
+            : base(sampleName, samplesDir ?? "test/test-applications/security", outputHelper)
+        {
+            httpClient = new HttpClient();
+        }
+
+        public async Task<MockTracerAgent> RunOnSelfHosted(bool enableSecurity, bool enableBlocking)
+        {
+            var agentPort = TcpPortProvider.GetOpenPort();
+            httpPort = TcpPortProvider.GetOpenPort();
+
+            var agent = new MockTracerAgent(agentPort);
+            await StartSample(agent.Port, arguments: null, aspNetCorePort: httpPort, enableSecurity: enableSecurity, enableBlocking: enableBlocking);
+            return agent;
+        }
+
+        public void Dispose()
+        {
+            if (process != null && !process.HasExited)
+            {
+                process.Kill();
+                process.Dispose();
+            }
+
+            if (httpClient != null)
+            {
+                httpClient.Dispose();
+            }
+        }
+
+        public async Task TestBlockedRequestAsync(MockTracerAgent agent, bool enableSecurity, HttpStatusCode expectedStatusCode, int expectedSpans, IEnumerable<Action<MockTracerAgent.Span>> assertOnSpans)
+        {
+            var mockTracerAgentAppSecWrapper = new MockTracerAgentAppSecWrapper(agent);
+            mockTracerAgentAppSecWrapper.SubscribeAppSecEvents();
+            Func<Task<(HttpStatusCode StatusCode, string ResponseText)>> attack = () => SubmitRequest("/Health/?arg=[$slice]");
```

Comment:
Should we be testing more than just this one endpoint? e.g. MVC controller (done), endpoint routed action, middleware, branched middleware, static file etc?

Reply:
Yes, but I'd prefer to do this after we merge, logged as: https://datadoghq.atlassian.net/browse/APPSEC-999